### PR TITLE
Update compiling_for_osx.rst

### PR DESCRIPTION
**Pull request by clemens-tolboom**
No sure 8 jobs is best for everyone but it boost compiling

### CODE REVIEW

**Comment by Calinou** [comment]
It's probably better to use a substitution that automatically detects the number of CPU threads:

```
scons -j$(sysctl -n hw.logicalcpu)
```

**Comment by clemens-tolboom** [comment]
@Calinou that's way better :)

File: development/compiling/compiling_for_osx.rst [hunk 1]
```diff
@@ -26,7 +26,7 @@ Start a terminal, go to the root dir of the engine source code and type:
 
 ::
 
-    user@host:~/godot$ scons platform=osx
+    user@host:~/godot$ scons platform=osx -j$(sysctl -n hw.logicalcpu)
```
Comment (Calinou) [inline development/compiling/compiling_for_osx.rst]: I also wonder if we should remove the shell placeholder to make it easier to copy-paste:

```suggestion
    scons platform=osx -j$(sysctl -n hw.logicalcpu)
```

This way, the text can be clicked 3 times to select the entire line (without any extraneous text).

Comment (clemens-tolboom) [reply development/compiling/compiling_for_osx.rst]: Done ... I reintroduced `--jobs` as that helps for scons novices like me.

Should I squash the 3 commits?

Comment (Calinou) [reply development/compiling/compiling_for_osx.rst]: It's OK, I can squash them myself.

**Comment by Calinou** [comment]
Thanks!

**Comment by clemens-tolboom** [comment]
Thanks too!

**Pull request merged**